### ISSUE TITLE
OOME when importing 17kb MachO binary

### ISSUE THREAD

**Issue by MarcSchoenefeld** [NationalSecurityAgency/ghidra#1488]
**Describe the bug**
Import the attached 17k binary (fuzzed variant of /usr/bin/true), and an uncaught  OOME (Java heap space) occurs.  

**To Reproduce**
Go To "Import File"
Import t41_eternal.bin

On Console this is shown: 

<details><summary>stack trace</summary>

```
ERROR Task Error: Import - Uncaught Exception: java.lang.OutOfMemoryError: Java heap space  (Task.java:133) 
java.lang.OutOfMemoryError: Java heap space
	at java.util.Arrays.copyOf(Arrays.java:3690) ~[?:?]
	at java.util.ArrayList.grow(ArrayList.java:235) ~[?:?]
	at java.util.ArrayList.grow(ArrayList.java:242) ~[?:?]
	at java.util.ArrayList.add(ArrayList.java:452) ~[?:?]
	at java.util.ArrayList.add(ArrayList.java:465) ~[?:?]
	at ghidra.app.util.bin.format.macho.Section.initSection(Section.java:96) ~[Base.jar:?]
	at ghidra.app.util.bin.format.macho.Section.createSection(Section.java:56) ~[Base.jar:?]
	at ghidra.app.util.bin.format.macho.commands.SegmentCommand.initSegmentCommand(SegmentCommand.java:91) ~[Base.jar:?]
	at ghidra.app.util.bin.format.macho.commands.SegmentCommand.createSegmentCommand(SegmentCommand.java:57) ~[Base.jar:?]
	at ghidra.app.util.bin.format.macho.commands.LoadCommandTypes.getLoadCommand(LoadCommandTypes.java:102) ~[Base.jar:?]
	at ghidra.app.util.bin.format.macho.MachHeader.parse(MachHeader.java:139) ~[Base.jar:?]
	at ghidra.app.util.bin.format.macho.MachHeader$$EnhancerByCGLIB$$bea828c2.CGLIB$parse$1(<generated>) ~[cglib-nodep-2.2.jar:?]
	at ghidra.app.util.bin.format.macho.MachHeader$$EnhancerByCGLIB$$bea828c2$$FastClassByCGLIB$$b2baa003.invoke(<generated>) ~[cglib-nodep-2.2.jar:?]
	at net.sf.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:215) ~[cglib-nodep-2.2.jar:?]
	at generic.continues.ContinuesInterceptor.intercept(ContinuesInterceptor.java:39) ~[Generic.jar:?]
	at ghidra.app.util.bin.format.macho.MachHeader$$EnhancerByCGLIB$$bea828c2.parse(<generated>) ~[cglib-nodep-2.2.jar:?]
	at ghidra.app.util.opinion.MachoPrelinkUtils.parsePrelinkXml(MachoPrelinkUtils.java:57) ~[Base.jar:?]
	at ghidra.file.formats.ios.prelink.PrelinkFileSystem.isValid(PrelinkFileSystem.java:79) ~[FileFormats.jar:?]
	at ghidra.formats.gfilesystem.factory.GFileSystemBaseFactory.probe(GFileSystemBaseFactory.java:66) ~[Base.jar:?]
	at ghidra.formats.gfilesystem.factory.FileSystemFactoryMgr.test(FileSystemFactoryMgr.java:233) ~[Base.jar:?]
	at ghidra.formats.gfilesystem.FileSystemService.isFileFilesystemContainer(FileSystemService.java:553) ~[Base.jar:?]
	at ghidra.plugin.importer.ImporterUtilities.showImportDialog(ImporterUtilities.java:172) ~[Base.jar:?]
	at ghidra.plugin.importer.ImporterUtilities.lambda$showImportDialog$0(ImporterUtilities.java:138) ~[Base.jar:?]
	at ghidra.plugin.importer.ImporterUtilities$$Lambda$645/0x0000000801500840.monitoredRun(Unknown Source) ~[?:?]
	at ghidra.util.task.TaskLauncher$2.run(TaskLauncher.java:119) ~[Docking.jar:?]
	at ghidra.util.task.Task.monitoredRun(Task.java:126) [Generic.jar:?]
	at ghidra.util.task.TaskRunner.lambda$startTaskThread$1(TaskRunner.java:94) [Docking.jar:?]
	at ghidra.util.task.TaskRunner$$Lambda$375/0x0000000801140440.run(Unknown Source) [Docking.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:830) [?:?]

```

</details>

in GUI:

Import - Uncaught Exception: java.lang.OutOfMemoryError: Java heap space
java.lang.OutOfMemoryError: Java heap space

---------------------------------------------------
Build Date: 2019-Dec-18 1306 EST
Ghidra Version: 9.1.1
Java Home: /Library/Java/JavaVirtualMachines/jdk-13.0.2.jdk/Contents/Home
JVM Version: Oracle Corporation 13.0.2
OS: Mac OS X 10.14.6 x86_64
Workstation: 192.168.x.x

**Expected behavior**
Display Macho format error and do not oome

**Comment by MarcSchoenefeld** [comment]
[ghidra_1488.zip](https://github.com/NationalSecurityAgency/ghidra/files/4127519/ghidra_1488.zip)

Fuzzed variant of /usr/bin/true (macos 10.14.6)

**Comment by astrelsky** [comment]
@MarcSchoenefeld may you wrap the stack trace in the following please?

````
<details><summary>stack trace</summary>
-required additional newline-
```
-stack trace-
```
</details>
````

**Comment by MarcSchoenefeld** [comment]
This seems to be problematic, https://github.com/NationalSecurityAgency/ghidra/blob/master/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/macho/Section.java#L96

**Comment by dev747368** [comment]
Issue isn't with macho, but a known issue with our low-level file io.  Its being worked on.

**Comment by dev747368** [comment]
Should be fixed by 4e1b743901b387518da9367c301860f0a1eef413.  Closing.
